src/freenas/etc/netcli

```diff
@@ -1619,17 +1629,43 @@ def main_menu():
             try:
                 if f not in [reset_root_pw, reset_factory_defaults, shell, reboot, shutdown]:
                     ensure_failover_is_disabled()
+                is_in_main_menu_prompt = False
                 f()
             except FailoverIsEnabledException:
                 print("You can't perform this action unless failover is administratively disabled")
                 print("Press Enter to return to console")
                 input()
 
 
+last_kernel_message = None
+
+
+def read_kmsg():
+    global last_kernel_message, main_menu_initialized
+    with open("/dev/kmsg") as f:
+        while True:
+            k = f.readline()
+            if main_menu_initialized:
+                last_kernel_message = time.monotonic()
+
+
+def press_enter_to_bring_console_setup_menu():
+    global last_kernel_message
+    while True:
+        if last_kernel_message is not None and time.monotonic() - last_kernel_message >= 3:
+            last_kernel_message = None
+            if is_in_main_menu_prompt:
+                print("Press Enter to bring Console setup menu.")
```

Comment:
Am I understanding correctly that this now gets printed to the console every 3 seconds?

Reply:
No, it only gets printed after we receive kernel messages (in 3 seconds after last message). Then `last_kernel_message` will be `None` until a new message is received